src/swaps/liqualityboost/LiqualityBoostSwapProvider.js

```diff
@@ -0,0 +1,290 @@
+import BN from 'bignumber.js'
+import { SwapProvider } from '../SwapProvider'
+import { unitToCurrency, assets } from '@liquality/cryptoassets'
+import { sha256 } from '@liquality/crypto'
+import { withLock, withInterval } from '../../store/actions/performNextAction/utils'
+import { prettyBalance } from '../../utils/coinFormatter'
+import { isERC20, getNativeAsset } from '@/utils/asset'
+import { createSwapProvider } from '../../store/factory/swapProvider'
+
+const slippagePercentage = 3
+
+class LiqualityBoostSwapProvider extends SwapProvider {
+  constructor (config) {
+    super(config)
+    this.liqualitySwapProvider = createSwapProvider(this.config.network, 'liquality')
+    this.oneinchSwapProvider = createSwapProvider(this.config.network, 'oneinchV3')
+  }
+
+  async getSupportedPairs () {
+    return []
+  }
+
+  async getQuote ({ network, from, to, amount }) {
+    if (isERC20(from) || !isERC20(to) || amount <= 0) return null
+    const toNativeAsset = getNativeAsset(to)
+    const quote = await this.liqualitySwapProvider.getQuote({ network, from, to: toNativeAsset, amount })
+    if (!quote) return null
+    const toNativeAssetQuantity = unitToCurrency(assets[toNativeAsset], quote.toAmount)
+    const finalQuote = await this.oneinchSwapProvider.getQuote({ network, from: toNativeAsset, to, amount: toNativeAssetQuantity.toNumber() })
+    if (!finalQuote) return null
+    return {
+      from,
+      to,
+      fromAmount: quote.fromAmount,
+      toAmount: finalQuote.toAmount,
+      toNativeAsset,
+      toNativeAssetAmount: quote.toAmount
```

Comment:
i find the `nativeAsset` naming a bit confusing. Possible to come up with something more accurate like `bridgeAsset` or `intermediaryAsset` ?